test/controllers/builds_controller_test.rb

```diff
@@ -98,4 +98,85 @@ def stub_git_reference_check(returns: false)
       end
     end
   end
+
+  as_a_viewer_project_deployer do
```

Comment:
Oh man, I know, couldn't come up with a better name. :)

Here I'd like to make it clear that this user is a User with 'Viewer' system level role, but he also has the 'Deployer' role on a specific project.

Would you prefer just `as_a_project_deployer`? Can you remember of a better name for this?